### PR TITLE
query_toks support for terrier.Retriever

### PR DESCRIPTION
**Pull request by cmacdonald**
{WIP}.

- [x] Need to be implemented for FBR too.

### CODE REVIEW

File: pyterrier/terrier/retriever.py [hunk 1]
```diff
@@ -301,13 +314,19 @@ def __setstate__(self, d):
...
-        if len(query) == 0:
-            warn("Skipping empty query for qid %s" % qid)
-            return []
 
-        srq = self.manager.newSearchRequest(qid, query)
-        
+        # row is a namedtuple, whose fields are exposed in _fields
+        query_toks = 'query_toks' in row._fields
+        if query_toks:
+            query = ''
```
Comment (seanmacavaney) [inline pyterrier/terrier/retriever.py:321]: `query = _querytoks2matchop(row.query_toks)`?

Comment (seanmacavaney) [reply pyterrier/terrier/retriever.py:321]: Oh, I see it's done below. I think it's clearer if you do it up here. Or at least give a comment saying that it's done below.

Comment (cmacdonald) [reply pyterrier/terrier/retriever.py:321]: aim is just to detect query_toks mode. Probably the variable should be "query_toks_present" to be clear its a bool flag.

**Comment by cmacdonald** [comment]
We need to update terrier-retrieval.rst documentation with something on pretokenised.

**Comment by cmacdonald** [comment]
I've added the relevant documentation, @seanmacavaney could you give a quick look.

**Review by seanmacavaney — APPROVED** [review]
applied two minor changes (one to documentation and one to a comment)

besides that lgtm

**Pull request merged**